assets/src/models/minischeduleData.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  Block,
+  Break,
+  Piece,
+  Run,
+  SignOnOff,
+  Time,
+  Trip,
+} from "../minischedule"
+import { RunId } from "../realtime"
+import { BlockId, DirectionId, RouteId, TripId } from "../schedule"
+
+interface RunData {
+  id: RunId
+  activities: (BreakData | PieceData)[]
+}
+
+interface BlockData {
+  id: BlockId
+  pieces: PieceData[]
+}
+
+interface BreakData {
+  break_type: string
+  start_time: Time
+  end_time: Time
+}
+
+interface PieceData {
+  run_id: RunId
+  block_id: BlockId
+  start: SignOnOffData
+  trips: TripData[]
+  end: SignOnOffData
+}
+
+interface SignOnOffData {
+  time: Time
+  place: string
+  "mid_route?": boolean
+}
+
+interface TripData {
+  id: TripId
+  block_id: BlockId
+  route_id: RouteId
+  headsign: string | null
+  direction_id: DirectionId | null
+  run_id: RunId | null
+}
+
+const isBreakData = (
+  activityData: BreakData | PieceData
+): activityData is BreakData => activityData.hasOwnProperty("break_type")
+
+export const runFromData = (runData: RunData): Run => ({
+  id: runData.id,
+  activities: runData.activities.map((activityData) =>
+    isBreakData(activityData)
+      ? breakFromData(activityData)
+      : pieceFromData(activityData)
+  ),
+})
+
+export const blockFromData = (blockData: BlockData): Block => ({
+  id: blockData.id,
+  pieces: blockData.pieces.map(pieceFromData),
+})
+
+export const breakFromData = (breakData: BreakData): Break => ({
```

Comment:
Not a big deal, but several of these functions are exported but not used outside of this file.